airbyte-integrations/connectors/source-jdbc/src/testFixtures/java/io/airbyte/integrations/source/jdbc/test/JdbcSourceAcceptanceTest.java

```diff
@@ -1061,4 +1061,17 @@ protected AirbyteMessage createStateMessage(final DbStreamState dbStreamState, f
     }
   }
 
+  public static void setEnv(final String key, final String value) {
+    try {
+      final Map<String, String> env = System.getenv();
+      final Class<?> cl = env.getClass();
+      final java.lang.reflect.Field field = cl.getDeclaredField("m");
+      field.setAccessible(true);
+      final Map<String, String> writableEnv = (Map<String, String>) field.get(env);
+      writableEnv.put(key, value);
+    } catch (final Exception e) {
+      throw new IllegalStateException("Failed to set environment variable", e);
+    }
+  }
```

Comment:
@subodh1810, if you want to set an environment variable in test, you can use the `@SetEnvironmentVariable` annotation:

See this [example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/PostgresSourceStrictEncryptAcceptanceTest.java#L37).

Should be easier than reflection.

Reply:
@tuliren ah nice! I was not aware of that! thank you! :D 
I will refactor my code to do it this way